pkg/operator/ceph/cluster/cluster.go

```diff
@@ -78,6 +78,21 @@ func newCluster(c *cephv1.CephCluster, context *clusterd.Context) *cluster {
 
 func (c *cluster) detectCephMajorVersion(image string, timeout time.Duration) (string, error) {
 	// get the major ceph version by running "ceph --version" in the ceph image
+	podSpec := v1.PodSpec{
+		Containers: []v1.Container{
+			{
+				Command: []string{"ceph"},
+				Args:    []string{"--version"},
+				Name:    "version",
+				Image:   image,
+			},
+		},
+		RestartPolicy: v1.RestartPolicyOnFailure,
+	}
+
+	// apply "all" placement
+	c.Spec.Placement.All().ApplyToPodSpec(&podSpec)
```

Comment:
What if we add a new method in this [placement](https://github.com/rook/rook/blob/master/pkg/apis/ceph.rook.io/v1/placement.go) file? Then we would add a new "version" tag in the `placement` section for the cluster crd. This option would be documented in the [placement](https://github.com/rook/rook/blob/0e4c447c76a74fc7b27fa6e127c69de51cd65fd6/Documentation/ceph-cluster-crd.md#placement-configuration-settings) crd doc.

Reply:
I don't quite understand the request. Are you suggesting adding a new Placement key, e.g. `PlacementKeyJobs`, and a corresponding function and configuration field?

Reply:
Yes, sounds like we're on the same page. This will allow the user to specify a different placement for jobs than the other ceph daemons. And calling it "jobs" sounds better than a specific "version" name just for this pod.

Reply:
@travisn Do "Ceph OSD prepare" Jobs also fall into that category then or (not sure if that issue is resolved) OSD prepare Jobs get the same placement as OSDs?

Reply:
If there's another job that follows the placement of an existing category, why not then have the Ceph Version job follow the mon placement?

Reply:
I don't know that I see a particular reason to have a placement just for jobs. I feel like that may be giving the user more flexibility at the expense of having to understand more things. I don't see why the `all` placement wouldn't apply to the detect version job; it's very lightweight. Using `all` should allow kubernetes more flexibility in where it can schedule the job if the some nodes are running low on resources.